### PR TITLE
feat: update e2e test version and ccm version

### LINKED ISSUE #139: Update upgrade e2e test
**What would you like to be added**:
We should move the base upgrade test from `v1.23.8` to `v1.24.3` and then the version to upgrade to from `v1.24.2` to `v1.25.0`

**Why is this needed**:
We need to stay up-to-date with Kubernetes versions

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
make 1.24 default version in test.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #139

### CODE REVIEW

File: test/e2e/config/e2e_conf.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ providers:
           - sourcePath: "../data/infrastructure-oci/v1beta1/metadata.yaml"
 
 variables:
-  KUBERNETES_VERSION: "v1.23.8"
+  KUBERNETES_VERSION: "v1.24.4"
```
Comment (joekr) [inline test/e2e/config/e2e_conf.yaml:76]: I left the upgrade path from 1.23 to 1.24 since I don't think CCM will support 1.25 yet.

**Review by shyamradhakrishnan — COMMENTED** [review]
we need to add the changes other than just the image in the addons template right?

**Comment by joekr** [comment]
# unit test

# e2e test
```
Ran 5 of 22 Specs in 1725.488 seconds
SUCCESS! -- 5 Passed | 0 Failed | 0 Pending | 17 Skipped
```

**Comment by joekr** [comment]
> we need to add the changes other than just the image in the addons template right?

@shyamradhakrishnan I have updated `cluster-template-oci-addons.yaml` with the same changes that are in the `test/e2e/data/ccm/ccm.yaml` file.

**Review by shyamradhakrishnan — APPROVED** [review]

**Pull request merged**